lms/djangoapps/course_api/api.py

```diff
@@ -170,19 +171,34 @@ def list_course_keys(request, username, role):
     """
     user = get_effective_user(request.user, username)
 
-    course_keys = CourseOverview.get_all_course_keys()
+    all_course_keys = CourseOverview.get_all_course_keys()
 
     # Global staff have access to all courses. Filter courses for non-global staff.
     if GlobalStaff().has_user(user):
-        return course_keys
-
-    return LazySequence(
-        (
-            course_key for course_key in course_keys
-            if has_access(user, role, course_key)
-        ),
-        est_len=len(course_keys)
-    )
+        return all_course_keys
+
+    if role == 'staff':
+        # This short-circuit implementation bypasses has_access() which we think is too slow for some users when
+        # evaluating staff-level course access for Insights.  Various tickets have context on this issue: CR-2487,
+        # TNL-7448, DESUPPORT-416, and probably more.
+        #
+        # This is a simplified implementation that does not consider org-level access grants (e.g. when course_id is
+        # blank).
+        filtered_course_keys = (
+            CourseAccessRole.objects.filter(user=user, role='staff')
+            .exclude(course_id__exact='')
+            .values_list('course_id', flat=True)
```

Comment:
So I think what might be happening is that the Django ORM is trying to prep a query to exclude blanks, but to do so, it's trying to build a `CourseKey('')` for comparison, which fails.

Reply:
I'll dig just a bit tonight, need to step away again for a bit.

Reply:
Okay, try `.exclude(CourseKeyField.Empty)`

Reply:
Committed, let's see what the CI says.

Reply:
No Dice. =(

Reply:
CourseKey needs to be imported.

Reply:
`from opaque_keys.edx.django.models import CourseKeyField`

Reply:
Sorry, should have mentioned that explicitly.

Reply:
#managersTryToCode! I'll fix that up.

Reply:
this seems plausible!